### ISSUE TITLE
1.6.20 new 's' Meta Set text protocol flag breaks validation

### ISSUE THREAD

**Issue by zenbones** [memcached/memcached#1045]
In 1.6.20 two flags were added to the protocol, 'N' for vivifying append/prepend entries, and 's' for returning the size of the entry. There's no problem for older clients with 'N', as absence of the flag produces exactly the previous behavior. However, the server is sending back 's' size information even when the 's' flag is not sent in the request. Pre 1.6.20 clients that strictly validate will strictly break, as they do not about this new 's' flag, and it shouldn't be there. Reading the docs, it seems to me that an 's' flag should be sent back in the response only if an 's' flag is sent in the request, like the 'c' flag. That would present no issues.

Not a big deal to upgrade the client in my case, as we own the code, but I believe this is a bug, and may hit other clients.

**Comment by dormando** [comment]
That sounds alarming, but I'm not seeing that behavior on a quick protocol test. Would you mind showing me a session exhibiting the behavior? IE:
```
ms foo 2   
hi
HD
```

I tried a few commands and don't see the 's' flag coming back unless I explicitly send it
```
ms foo 2 MA
ko
HD
mg foo 
HD
mg foo v
VA 4
hiko
ms foo 2 MA s
zo
HD s6
mg foo v
VA 6
hikozo
```

**Comment by dormando** [comment]
Looking over the code and I don't see how it's possible for the 's' flag to reflect unless it was initially sent.

Checked to see if it was somehow leaking into the response buffer somehow but I don't see an obvious way for that either. If you have a reproducible test command I can probably pin it pretty quickly via debugger.

**Comment by zenbones** [comment]
Sorry, yes, I can run a test. My driver does not send the s flag, but does receive it, so let me get that for you. Sorry about not checking the replies here for a few days.

**Comment by zenbones** [comment]
ms c2Vjb25k 13 b ME c
<bytes>

The byte array is...
[109, 115, 32, 99, 50, 86, 106, 98, 50, 53, 107, 32, 49, 51, 32, 98, 32, 77, 69, 32, 99, 13, 10, -84, -19, 0, 5, 116, 0, 6, 118, 97, 108, 117, 101, 50, 13, 10]

And the response is...
HD c74 s13

**Comment by dormando** [comment]
Awesome thanks! That saved me some time as I was just about to try to figure this out again.

I boiled it down to this:
```
ms a 2 c
hi
HD c4 s2
```

Looks like the 'c' flag is somehow causing s to return. I'll make sure the test coverage is stricter and get this fixed in the next day or two; sorry for the trouble and thanks again for reaching out.

**Comment by dormando** [comment]
Fixed here: https://github.com/memcached/memcached/commit/2315e0c4c897ebe4c0fcf5d9fdcaaf57032bf0e6

New release should be in a day or two, as I'm packing in pent up bugfixes :(

The tests for meta protocol are in fact fairly long, but "ms with c" was never explicitly tested somehow. I made that work fairly late in the development process and missed it on a test audit. Ironically the ms N/s flags did have an outside code review for once...

Thanks again for your report and sorry for the inconvenience. I typically leave these issues open until a release tarball is cut.
